com.avaloq.tools.ddk.xtext.generator/src/com/avaloq/tools/ddk/xtext/generator/ecore/GeneratorAdapterDescriptor.java

```diff
@@ -0,0 +1,148 @@
+        @Override
+        protected void createImportManager(final String packageName, final String className) {
+          importManager = new ImportManagerHack(packageName, typeMapper);
+          importManager.addMasterImport(packageName, className);
+          if (generatingObject != null) {
+            ((GenBase) generatingObject).getGenModel().setImportManager(importManager);
+          }
+        }
+      };
+    }
+
+    @Override
+    public Adapter createGenModelAdapter() {
+      if (genModelGeneratorAdapter == null) {
+        genModelGeneratorAdapter = new GenModelGeneratorAdapter(this) {
+          @Override
+          protected void createImportManager(final String packageName, final String className) {
+            importManager = new ImportManagerHack(packageName, typeMapper);
+            importManager.addMasterImport(packageName, className);
+            if (generatingObject != null) {
+              ((GenBase) generatingObject).getGenModel().setImportManager(importManager);
+            }
+          }
+
+          /**
+           * We handle this one on our own, see
+           * {@link org.eclipse.xtext.generator.ecore.CvsIdFilteringGeneratorAdapterFactoryDescriptor.IdFilteringGenModelGeneratorAdapterFactory# createGenModelAdapter()}
+           * . Note that the original source comment vanishes in de-compiled Java code.
+           */
+          @Override
+          protected void generateModelPluginProperties(final GenModel genModel, final org.eclipse.emf.common.util.Monitor monitor) {
+            // Do nothing.
+          }
+        };
+      }
+      return genModelGeneratorAdapter;
+    }
+
+    @Override
+    public Adapter createGenPackageAdapter() {
+      return new IdFilteringGenPackageAdapter(this) {
+        @Override
+        protected void createImportManager(final String packageName, final String className) {
+          importManager = new ImportManagerHack(packageName, typeMapper);
+          importManager.addMasterImport(packageName, className);
+          if (generatingObject != null) {
+            ((GenBase) generatingObject).getGenModel().setImportManager(importManager);
+          }
+        }
+      };
+    }
+  }
+
+  private final Function<String, String> typeMapper;
+  private final Set<String> loggedTypeMappings;
+  private static final Logger LOGGER = Logger.getLogger(GeneratorAdapterDescriptor.class);
+
+  protected GeneratorAdapterDescriptor(final Set<String> loggedTypeMappings, final Function<String, String> typeMapper) { // NOPMD
+    this.typeMapper = typeMapper;
+    this.loggedTypeMappings = loggedTypeMappings;
+  }
+
+  @Override
+  public GeneratorAdapterFactory createAdapterFactory() {
+    return new CustomImplAwareGeneratorAdapterFactory();
+  }
+
+  /**
+   * Hacked EMF ImportManager which returns the name of the custom classes as returned by the {@link TypeMapper}.
+   */
+  protected class ImportManagerHack extends ImportManager {

```

Comment:
Why *Hack?

Reply:
No idea - This file was created from code extracted from CustomClassEcoreGeneratorComponent, so most of the code and comments are from there. The class name was probably chosen to match the inner class org.eclipse.emf.mwe2.ecore.EcoreGenerator.ImportManagerHack which is also a wrapper around ImportManager, but I don't think there is a reason we have to keep the name.